deployment-bundle-steps.md

```diff
@@ -0,0 +1,135 @@
+# How to setup a project with deployment to S3
+
+1. Create S3 bucket for production / staging (your-bucket-name-production & your-bucket-name-staging). (If using cloudfront, see [optional settings](#optional-settings))
+
+2. Log into [circleci.com](https://circleci.com/) and add your project. Set up the following credentials: (see other members of the team for these values).
+  - Environment Variables
+    - GHUSER
+    - GHPASS
+    - GRAPHITE_API_KEY
+    - GRAPHITE_ENDPOINT_PREFIX
+    - CLOUDFRONT_DISTRIBUTION_ID (if using cloudfront for your production bundle)
+  - AWS CodeDeploy
+    - Access Key Id
+    - Secret Access Key
+
+3. Create relevant files for your project:
+  - [circle.yml](#circleyml)
+  - [scripts/deployS3.sh](#deploys3sh)
+  - [istanbul.yml](#istanbulyml)
+
+4. Add [scripts](#npm-scripts) to your package.json
+
+5. Add [devDependencies](#devdependencies) to your package.json
+
+6. Make sure your tests & linting pass. Merge to staging, see it deploy!
+
+---
+
+##### Optional Settings
+
+1a. (optional) Setup cloud front on the production bucket & add the following to the end of your `deployS3.sh` script:
+```bash
+if [ $1 == 'production' ]; then
+  aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths /\*
+fi
+```
+
+---
+
+#### Examples
+
+##### Npm scripts:
+```json
+"scripts": {
+  "build": "webpack --config ./webpack.config.js",
+  "start": "WEBPACK_DEV=true webpack-dev-server --config ./webpack.config.js --content-base ./demo/ --inline --progress",
+  "lint": "make-up src test -i eslint",
+  "ci": "TEST_REPORT_DIRECTORY=${1:-unit}; istanbul cover _mocha --  test/ -R xunit > $CIRCLE_TEST_REPORTS/$TEST_REPORT_DIRECTORY/results.xml",
+  "prerelease": "node_modules/@holidayextras/deployment-helpers/nodeApps/preRelease.sh",
```

Comment:
I'm not sure how I feel about this... how do we enforce semver... would we automatically update as a patch unless told otherwise? What happens if dev's become dependent on the automation? Do we end up having breaking changes being deployed under a patch revision?